uni/lib/controller/local_storage/app_shared_preferences.dart

```diff
@@ -117,6 +120,20 @@ class AppSharedPreferences {
     return prefs.setInt(themeMode, (themeIndex + 1) % 3);
   }
 
+  static Future<void> setLocale(AppLocale appLocale) async {
+    final prefs = await SharedPreferences.getInstance();
+    await prefs.setString(locale, appLocale.name);
+  }
+
+  static Future<AppLocale> getLocale() async {
+    final prefs = await SharedPreferences.getInstance();
+    final appLocale =
+        prefs.getString(locale) ?? Platform.localeName.substring(0, 2);
+
+    return AppLocale.values
+        .firstWhere((e) => e.toString() == 'AppLocale.$appLocale');
+  }
```

Comment:
- So if `.name` returns the string representation of the variant without any prefix,how does checking against the prefixed string below works?
- The default should be English, not whatever language the user has set in their phone.
- firstWhere will throw if it does not find something. Pass `orElse` to return English

Reply:
We are checking against `AppLocale.values` which return `[AppLocale.en, AppLocale.pt]`.

Reply:
Hm sure, what about the other comments?

Reply:
I guess the default could be whatever language the user has in their phone, but we must check with `orElse` if we support that language, otherwise we return English.

Reply:
Sure, seems fine to me